### PR TITLE
[BAEL-6708] Add Cartesian Product

### PR DESCRIPTION
**Pull request by rajat-garg**
Calculate Cartesian Product using various approaches

### CODE REVIEW

File: core-java-modules/core-java-collections-set-2/src/test/java/com/baeldung/cartesianproduct/CartesianProductUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+public class CartesianProductUnitTest {
+    private CartesianProduct cp = new CartesianProduct();
+    List<List<Object>> sets = Arrays.asList(
+        Arrays.asList(10, 20),
+        Arrays.asList("John", "Jack"),
+        Arrays.asList('I', 'J')
+    );
+
+    @Test
+    public void shouldCalculateCartesianProductUsingStreams() {
```
Comment (vinipx) [inline core-java-modules/core-java-collections-set-2/src/test/java/com/baeldung/cartesianproduct/CartesianProductUnitTest.java:22]: Please, follow the guidelines for [writing tests](https://docs.google.com/document/d/1iIeSPuYFG-jEtB8UbcxQnn2AI3OR2LWlnisNo2NUCL8/edit) for test method naming and content

Comment (vinipx) [reply core-java-modules/core-java-collections-set-2/src/test/java/com/baeldung/cartesianproduct/CartesianProductUnitTest.java:22]: and apply for all test methods

File: core-java-modules/core-java-collections-set-2/src/main/java/com/baeldung/cartesianproduct/CartesianProduct.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.Sets;
+
+public class CartesianProduct {
+    // Responsible for calculating Cartesian Product using iterative approach.
```
Comment (vinipx) [inline core-java-modules/core-java-collections-set-2/src/main/java/com/baeldung/cartesianproduct/CartesianProduct.java:12]: let's remove the comments and make sure the method naming reflects its intent/objective

**Review by vinipx — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-collections-set-2/src/main/java/com/baeldung/cartesianproduct/CartesianProduct.java (hunk 2 above)
Comment (rajat-garg) [reply core-java-modules/core-java-collections-set-2/src/main/java/com/baeldung/cartesianproduct/CartesianProduct.java]: Done

File: core-java-modules/core-java-collections-set-2/src/test/java/com/baeldung/cartesianproduct/CartesianProductUnitTest.java (hunk 1 above)
Comment (rajat-garg) [reply core-java-modules/core-java-collections-set-2/src/test/java/com/baeldung/cartesianproduct/CartesianProductUnitTest.java]: Somehow I missed this information, thank you for pointing it out. It's fixed now.

**Review by vinipx — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-collections-set-2/pom.xml [hunk 3]
```diff
@@ -20,6 +20,17 @@
...
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.7.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>31.0.1-jre</version>
```
[the remark this answers is not in the document]
Comment (rajat-garg) [reply core-java-modules/core-java-collections-set-2/pom.xml]: Done

**Pull request merged**